src/psyclone/dynamo0p3.py

```diff
@@ -8193,6 +8247,32 @@ def mesh(self):
         '''
         return self._mesh_properties
 
+    @property
+    def all_updates_are_writes(self):
+        '''
+        :returns: true if all of the arguments updated by this kernel have \
+                  GH_WRITE access, false otherwise.
+        :rtype: bool
+
+        '''
+        if self.is_intergrid:
+            # This is not a special kernel
+            return False
+        accesses = set(arg.access for arg in self.args)
+        all_writes = AccessType.all_write_accesses()
+        all_writes.remove(AccessType.WRITE)
+        if not accesses.intersection(set(all_writes)):
+            return True
+        return False
+
+    @property
+    def accesses_annexed_dofs(self):
+        '''
+        :returns: true if this kernel accesses annexed dofs, false otherwise.
```

Comment:
I would suggest "True" and "False" (start with upper case) for consistency with other docstrings.

Reply:
done.